### DISCUSSION TITLE
Load Nested Relationships and orderColumn customization

### DISCUSSION THREAD

**Discussion by ybdendreo** [yajra/laravel-datatables#2905] [General]
Hello,

On the first hand, I want to thanks you for your great job 👏🏼 

I have a question on the nested relationships, as you can see on my screenshot, I load my relationships with a subquery but at this step, the method `isNotEagerLoaded` return me `true` (Is that a first problem ? if i want that the method return false I need to use `with('module.categorie')`) and I can use the `orderColumn` method to customize my order.

![code](https://user-images.githubusercontent.com/99175520/202718442-e31cd85d-8322-45ce-ac43-daafd77e4fd8.png)

Now considere that I want to customize the order column of the `module.categorie` relation, if your method `isNotEagerLoaded` return me `true` my `orderColumn` method is not supported, Do you have a solution for me ?

**Comment by yajra** [comment]
Looks like `isNotEagerLoaded` method needs improvements when using a closure.  Maybe a workaround is to use the custom `column.name` trick. Something like:

**JS**

```js
columns: [
  {data: 'module.categorie.name', name: 'module_categorie'}
]
```

**PHP**

```php
->orderColumn('module_categorie', fn($q) => ...)
->filterColumn('module_categorie', fn($q) => $q->whereHas('module.categorie', fn($q) => ...)
```

**Reply by ybdendreo** [reply]
Thank you for your answer @yajra :)
